core/src/main/java/org/ehcache/core/spi/service/LocalPersistenceService.java

```diff
@@ -17,22 +17,63 @@
 package org.ehcache.core.spi.service;
 
 import org.ehcache.CachePersistenceException;
-import org.ehcache.spi.persistence.PersistableResourceService;
+import org.ehcache.spi.service.MaintainableService;
+
+import java.io.File;
 
 /**
- * Service to provide persistence context to caches requiring it.
- * <P>
- * Will be used by caches with a disk store, whether or not the data should survive a program restart.
- * <P/>
+ * Service that provides unique local persistence spaces (and safe sub spaces) to any service that requires it
+ * under the local root directory.
  */
-public interface LocalPersistenceService extends PersistableResourceService {
+public interface LocalPersistenceService extends MaintainableService {
 
   /**
-   * Creates a new persistence context within the given space.
+   * Creates a logical safe directory space for the owner and returns an identifying space Id.
+   *
+   * @param owner Service owner that owns the safe space.
+   * @param name Identifying name for the space.
    *
-   * @param identifier space to create within
-   * @param name name of the context to create
-   * @return a {@link FileBasedPersistenceContext}
+   * @return Opaque Identifier that can be used to identify the safe space.
+   */
+  SafeSpaceIdentifier createSafeSpaceIdentifier(String owner, String name);
+
+  /**
+   * Creates or finds the safe space represented by {@code safeSpaceId} in the underlying physical space.
+   *
+   * @param safeSpaceId Identifier to the created logical space on which the physical space needs to be created
+   * @throws CachePersistenceException If the space cannot be created or found, due to system errors
+   */
+  void createSafeSpace(SafeSpaceIdentifier safeSpaceId) throws CachePersistenceException;
+
+  /**
+   * Destroys the safe space.
+   *
+   * @param safeSpaceId Safe space identifier.
+   * @param verbose Log more information.
+   */
+  void destroySafeSpace(SafeSpaceIdentifier safeSpaceId, boolean verbose);
+
+  /**
+   * Destroys all safe spaces provided to this owner.
+   *
+   * @param owner owner of safe spaces.
+   */
+  void destroyAll(String owner);
+
+  /**
+   * Identifier to the logical safe space
    */
-  FileBasedPersistenceContext createPersistenceContextWithin(PersistenceSpaceIdentifier<?> identifier, String name) throws CachePersistenceException;
-}
+  interface SafeSpaceIdentifier {
```

Comment:
Isn't SafeSpaceIdentifier consistent with the rest of the calls (such as createSafeSpaceId..etc). So should I rename every method in the interface to FileSpace from SafeSpace?.

Isn't FileSpace pleonastic? or another way to look at it is space for file (which is directory). Since the service does a bit more than providing a directory (such as locking to prevent accidental use of the same directory under different cache managers and validating and unique naming etc.) thought SafeSpace conveys the meaning better.